web/src/pages/callback.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import { Spin } from "antd";
+import * as setting from '../Setting'
+import * as Backend from '../Backend'
+
+class Callback extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classes: props,
+      addition: props.match.params.addition,
+      state: "",
+      code: ""
+    };
+    const params = new URLSearchParams(this.props.location.search);
+    this.state.code = params.get("code");
+    this.state.state = params.get("state");
+    setting.initFullClientUrl();
+    Backend.initServerUrl();
+  }
+
+  getAuthenticatedInfo() {
+    let redirectUrl;
+    redirectUrl = `${setting.ClientUrl}/callback/${this.state.addition}`;
+    Backend.githubLogin(this.state.code, this.state.state, redirectUrl, this.state.addition)
+      .then((res) => {
+        if (res.status === "fail") {
+          setting.showMessage("error", res?.msg);
+        }
```

Comment:
here looks lost return.

Reply:
The callback page only serves as a transition to the home page, so here not needs to return.

Reply:
@nodece Thanks for the advice. Please review.